crates/wal-protocol/src/control.rs

```diff
@@ -16,8 +16,56 @@ use restate_types::GenerationalNodeId;
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct AnnounceLeader {
-    /// The "generational" node id of the processor that is being announced as leader.
-    /// This is used because the header only contains the plain id.
-    pub node_id: GenerationalNodeId,
+    // todo: Remove once we no longer need to support rolling back to 1.0
+    pub node_id: Option<GenerationalNodeId>,
     pub leader_epoch: LeaderEpoch,
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::control::AnnounceLeader;
+    use bytes::BytesMut;
+    use restate_types::identifiers::LeaderEpoch;
+    use restate_types::storage::StorageCodec;
+    use restate_types::{flexbuffers_storage_encode_decode, GenerationalNodeId};
+
+    #[derive(Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+    struct OldAnnounceLeader {
+        pub node_id: GenerationalNodeId,
+        pub leader_epoch: LeaderEpoch,
+    }
+
+    flexbuffers_storage_encode_decode!(AnnounceLeader);
+    flexbuffers_storage_encode_decode!(OldAnnounceLeader);
```

Comment:
Fair point. The reason why I am leaning a bit towards keeping it is because it documents the older versions of this message. If the next change to `AnnounceLeader` happens we will not accidentally forget about the existence of `OldAnnounceLeader` with this test.